src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/TrustsDataController.java

```diff
@@ -0,0 +1,154 @@
+    public TrustsDataController(
+            final PrivateDataRetrievalService privateDataRetrievalService,
+            final OverseasEntitiesService overseasEntitiesService
+    ) {
+        this.privateDataRetrievalService = privateDataRetrievalService;
+        this.overseasEntitiesService = overseasEntitiesService;
+
+        this.privateDataRetrievalService.setHashHelper(hashHelper);
+    }
+
+
+    private Map makeLogMap(
+            String transactionId,
+            String overseasEntityId
+    ){
+        return Map.of(
+                Constants.OVERSEAS_ENTITY_ID_KEY, overseasEntityId,
+                Constants.TRANSACTION_ID_KEY, transactionId
+        );
+    }
+
+
+    @GetMapping("/{trust_id}/corporate-trustees")
+    public ResponseEntity<PrivateCorporateTrusteeListApi> getCorporateTrusts(
+            @PathVariable(Constants.TRANSACTION_ID_KEY) String transactionId,
+            @PathVariable(Constants.OVERSEAS_ENTITY_ID_KEY) String overseasEntityId,
+            @PathVariable(Constants.TRUST_ID) String trustId,
+            @RequestHeader(value = Constants.ERIC_REQUEST_ID_KEY) String requestId)
+            throws ServiceException {
+
+        logMap = makeLogMap(transactionId, overseasEntityId);
+        String companyNo = getCompanyNumber(overseasEntityId,requestId);
+        if(companyNo == null){
+            return ResponseEntity.notFound().build();
+        }
+        Callable<PrivateCorporateTrusteeListApi> function = () -> privateDataRetrievalService.getCorporateTrustees(trustId, companyNo);
+        return checkSubmissionDto(function, requestId, "corporate trustee");
+
+    }
+
+    public String getCompanyNumber(
+            String overseasEntityId,
+            String requestId
+    ) throws ServiceException {
+
+        ApiLogger.infoContext(requestId,
+                "Calling Overseas Entities Service to retrieve private trust data for overseas entity "
+                        + overseasEntityId, logMap);
+        final var submissionDtoOptional = overseasEntitiesService.getOverseasEntitySubmission(
+                overseasEntityId);
+
+        if (!submissionDtoOptional.isPresent()) {
+            ApiLogger.errorContext(requestId,
+                    "Could not find overseas entity submission for overseas entity "
+                            + overseasEntityId, null, logMap);
+            return null;
+        }
+
+        final var submissionDto = submissionDtoOptional.get();
+
+        if (!submissionDto.isForUpdate()) {
+            throw new ServiceException(
+                    "Submission for overseas entity details must be for update");
+        }
+        if (!isRoeUpdateEnabled) {
+            throw new ServiceException(
+                    "ROE Update feature must be enabled for get overseas entity details");
+        }
+
+        return submissionDto.getEntityNumber();
+    }
+
+
+    private <U extends Hashable, T extends PrivateDataList<U>> ResponseEntity<T> checkSubmissionDto(
```

Comment:
Didn't we agree to change this to 
```suggestion
    private <U extends Hashable, T extends PrivateDataList<U>> ResponseEntity<T> retrievePrivateTrustData(
```
? Test names will need to be updated too